t/lib/OpenQA/Test/FullstackUtils.pm

```diff
@@ -163,7 +163,7 @@ sub wait_for_developer_console_like {
         }
 
         wait_for_ajax(msg => $message_regex);
-        javascript_console_has_no_warnings_or_errors($js_erro_check_suffix) or return;
+        ok(javascript_console_has_no_warnings_or_errors($js_erro_check_suffix), 'No unexpected js warnings') or return;
```

Comment:
here I still see the problem of using Test::More calls in a sub-function where the reporting scope doesn't fit. So better avoid "ok" and other functions like that. Can you just call "die" or simply use the return?

Reply:
This function already relies on `pass` and `fail`. That said I could leave this to `return` only